### ISSUE TITLE
add test data dumper to each source

### ISSUE THREAD

**Issue by nlwashington** [monarch-initiative/dipper#91]
This refactor includes several steps, including:

1.  each source must have a test-data mode that will dump only test data into a special file based on a flag
2.  test-data will dump to a separate/additional test-dump data file.  this should go into out/test/*_test.ttl
3.  unit tests be written for each source that will make sure that a test file is dumped and is parsable
4.  add documentation in the Source class that describes the need for test data.
5.  add documentation in the dipper class that explains our testing data strategy; perhaps how we choose subsets of data?

here, we should think critically about what data we choose for the test set and why.  we should make sure that the choice of test data is designed so that we can test specific things downstream (certain kinds of graph queries), and make notes of that in each source's documentation.

we should be sure to include some data from each source that as some thread that ties it across data sources, with some additional data to test the breadth of data modeling that we're doing.  that should include at least one instance/class that is instantiated per type, and at least one of each kind of triple that is generated.

**Comment by nlwashington** [comment]
as of merge commit dd02236, the following has been implemented:

* each source must have a test-data mode...
  * This is now available using a ```--test_only``` flag
* test-data will dump to a separate/additional test-dump data file. 
  * Each data source will have a shadow dataset called source_test.ttl.
  * If you are running the unit tests, it will go into the tests/out/sourcename_test.ttl, otherwise it will go in just plain /out/sourcename_test.ttl.  This will happen in parallel to the full file (with any count limits or other filters applied).
* unit tests be written for each source
  * each source now has a unit test under ```tests```, named ```test_sourcename.py```. The testing classes defined in that file are also added to the source python code in a testsuite.  
  * the graph load and owltools load tests are part of the standard unittest suite.  new sources should create a unittest script.
  * most sources still need to have source-specific tests written (as in, sparql-type queries)
* i have started documentation in the Source class in a new [README](https://github.com/monarch-initiative/dipper/blob/master/dipper/sources/README.md) that describes the need for test data, and some general information about what to add.  this will be built out more in another ticket.
* add documentation in the dipper class that explains our testing data strategy...
  * this will get added as another ticket.
